Adding MacOs getting started (#239)
dotnet/spark: docs/getting-started/macos-instructions.md

# Getting Started with Spark .NET on MacOS

These instructions will show you how to run a .NET for Apache Spark app using .NET Core on MacOSX.

## Pre-requisites

- Download and install the following: **[.NET Core 2.1 SDK](https://dotnet.microsoft.com/download/dotnet-core/2.1)** 
- Download and install Java using 

>   ```brew cask install java```

Reviewer: ```suggestion
    ```shell
    brew cask install java
    ```
```

> - Download and install Scala using ```brew install scala```

Reviewer: ```suggestion
- Download and install Scala using
    ```shell
    brew install scala
    ```
```

> - Download and install Apache Spark using ```brew install apache-spark```

Reviewer: ```suggestion
- Download and install Apache Spark using
    ```shell
    brew install apache-spark
    ```
```

    
- Download and install **[Microsoft.Spark.Worker](https://github.com/dotnet/spark/releases)** release:
    - Select a **[Microsoft.Spark.Worker](https://github.com/dotnet/spark/releases)** release from .NET for Apache Spark GitHub Releases page and download into your local machine (e.g., `/bin/Microsoft.Spark.Worker-0.4.0`).
    - **IMPORTANT** Create a new environment variable using ```export DOTNET_WORKER_DIR <your_path>``` and set it to the directory where you downloaded and extracted the Microsoft.Spark.Worker (e.g., `/bin/Microsoft.Spark.Worker-0.4.0`).

## Authoring a .NET for Apache Spark App

> - Create a new Console Appliction using  `dotnet new console -n HelloSpark`

Reviewer: ```suggestion
- Use the `dotnet` CLI to create a console application.
    ```shell
    dotnet new console -o HelloSpark
    ```
```

> - Install ```Microsoft.Spark``` Nuget package into the solution by executing the following command ```dotnet add package Microsoft.Spark``` into the project folder.

Reviewer: ```suggestion
- Install `Microsoft.Spark` Nuget package into the project from the [spark nuget.org feed](https://www.nuget.org/profiles/spark) - see [Ways to install Nuget Package](https://docs.microsoft.com/en-us/nuget/consume-packages/ways-to-install-a-package)
    ```shell
    cd HelloSpark
    dotnet add package Microsoft.Spark
    ```
```

> - Write the following code into `Program.cs`:

Reviewer: ```suggestion
- Replace the contents of the `Program.cs` file with the following code:
```

    ```csharp
    using Microsoft.Spark.Sql;

    namespace HelloSpark
    {
        class Program
        {
            static void Main(string[] args)
            {
                var spark = SparkSession.Builder().GetOrCreate();
                var df = spark.Read().Json("people.json");
                df.Show();
            }
        }
    }
    ```

> - Build the solution using ```dotnet build```

Reviewer: ```suggestion
- Use the `dotnet` CLI to build the application:
    ```shell
    dotnet build
    ```
```

## Running your .NET for Apache Spark App
- Open your terminal and navigate into your app folder:

>     ```

Reviewer: ```suggestion
    ```shell
```

# Getting Started with Spark .NET on MacOS

These instructions will show you how to run a .NET for Apache Spark app using .NET Core on MacOSX.

## Pre-requisites

- Download and install the following: **[.NET Core 2.1 SDK](https://dotnet.microsoft.com/download/dotnet-core/2.1)** 
- Download and install Java using 
    ```shell

>     brew cask install java

Reviewer: We should be installing java8.  Doesn't the current command run [this](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/java.rb) which installs java12?

Reviewer: Also since homebrew isn't shipped with MacOS by default and is a 3rd party package management solution, I would add some note somewhere regarding this and that these `brew` commands assumes it's installed.

    ```
- Install `Microsoft.Spark` Nuget package into the project from the [spark nuget.org feed](https://www.nuget.org/profiles/spark) - see [Ways to install Nuget Package](https://docs.microsoft.com/en-us/nuget/consume-packages/ways-to-install-a-package)
    ```shell
    cd HelloSpark
    dotnet add package Microsoft.Spark
    ```
- Replace the contents of the `Program.cs` file with the following code:
    ```csharp
    using Microsoft.Spark.Sql;

    namespace HelloSpark
    {
        class Program
        {
            static void Main(string[] args)
            {
                var spark = SparkSession.Builder().GetOrCreate();
                var df = spark.Read().Json("people.json");
                df.Show();
            }
        }
    }
    ```
- Use the `dotnet` CLI to build the application:
    ```shell
    dotnet build
    ```

## Running your .NET for Apache Spark App
- Open your terminal and navigate into your app folder:
    ```shell
    cd <your-app-output-directory>
    ```
- Create `people.json` with the following content:
    ```json
    {"name":"Michael"}
    {"name":"Andy", "age":30}
    {"name":"Justin", "age":19}
    ```
- Run your app
    ```

>     spark-submit `

Reviewer: Multiline commands on Mac require  \`s ?   I thought they would be similar to linux, `\`

    --class org.apache.spark.deploy.dotnet.DotnetRunner `
    --master local `
    microsoft-spark-2.4.x-<version>.jar `
    dotnet HelloSpark.dll

>     ```

Reviewer: We need to add a note here.

    **Note**: This command assumes you have downloaded Apache Spark and added it to your PATH environment variable to be able to use `spark-submit`, otherwise, you would have to use the full path (e.g., `~/spark/bin/spark-submit`).

# Getting Started with Spark .NET on MacOS

These instructions will show you how to run a .NET for Apache Spark app using .NET Core on MacOSX.

## Pre-requisites

- Download and install the following: **[.NET Core 2.1 SDK](https://dotnet.microsoft.com/download/dotnet-core/2.1)** 
- Download and install brew package manager using: 
    ```shell
    /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
    ```
- Download and install Java using 
    ```shell

>     brew tap caskroom/versions

Reviewer 2: when I run this I get the following:
`Error: caskroom/versions was moved. Tap homebrew/cask-versions instead.`

Should we just have a link to Java SDK and Spark binaries similar to other OS instructions so that we don't depend on `brew` behavior? (also, we don't have to worry about spark version installed by `brew install apache-spark`.)
